docs/tutorials/data_provider.md

```diff
@@ -80,61 +80,45 @@ use icu_provider::DataLocale;
 use icu_provider::hello_world::HelloWorldFormatter;
 use icu::locid::locale;
 use icu::locid::subtags::Language;
-use std::path::PathBuf;
+use std::path::Path;
 use writeable::Writeable;
 
-// Our growable data provider will be a fallback-enabled ForkByKeyProvider forking between a
-// BlobDataProvider with static core data and a MultiForkByErrorProvider with runtime-loaded data.
-type GrowableDataProvider = LocaleFallbackProvider<
-    ForkByKeyProvider<
-        BlobDataProvider,
-        MultiForkByErrorProvider<FsDataProvider, MissingLocalePredicate>
-    >
->;
-
 // Create the empty MultiForkByErrorProvider:
-let multi_lang_provider = MultiForkByErrorProvider::new_with_predicate(
+let mut provider = MultiForkByErrorProvider::new_with_predicate(
     vec![],
     MissingLocalePredicate
 );
 
-// Put together the rest of the data pipeline:
-static CORE_DATA: &[u8] = include_bytes!("../../../provider/adapters/tests/data/fallback.postcard");
-let provider = BlobDataProvider::try_new_from_static_blob(CORE_DATA)
-    .expect("Core data should be valid");
-let provider = ForkByKeyProvider::new(provider, multi_lang_provider);
-let mut provider: GrowableDataProvider = LocaleFallbackProvider::try_new_with_buffer_provider(provider)
-    .expect("Fallback data is in core data");
-
-// Helper function to add data into the GrowableDataProvider on demand:
+// Helper function to add data into the growable provider on demand:
 let mut get_hello_world_formatter = |loc: &DataLocale| {
     // Try to create the formatter a first time with data that has already been loaded.
     if let Ok(formatter) = HelloWorldFormatter::try_new_with_buffer_provider(&provider, loc) {
         return formatter;
     }
 
     // We failed to create the formatter. Load more data for the language.
-    let mut path_buf: PathBuf = "../../provider/adapters/tests/data/langtest".parse().unwrap();
-    path_buf.push(&*loc.language().write_to_string());
+    let path_buf = 
+        Path::new("../../provider/adapters/tests/data/langtest")
+        .join(&*loc.language().as_str());
     let lang_provider = match FsDataProvider::try_new(&path_buf) {
         Ok(p) => p,
         Err(e) => panic!("Language not available? {:?}", e)
     };
     println!("Successfully loaded: {:?}", loc);
 
-    // Add the data to the GrowableDataProvider and try creating the formatter a second time.
-    provider.inner_mut().inner_mut().1.push(lang_provider);
+    // Add the data to the growable provider and try creating the formatter a second time.
+    provider.push(lang_provider);
     HelloWorldFormatter::try_new_with_buffer_provider(&provider, loc)
         .expect("Language data should now be available")
```

Comment:
non-blocking: still not a fan of demonstrating bundling by language tag, as `zh` and `zh-Hant` should always be separate language packs

Reply:
I added an explainer paragraph with disclaimers.